build.sbt

```diff
@@ -394,6 +394,10 @@ val mimaFilterSettings = Seq {
     ProblemFilters.exclude[MissingTypesProblem]("scala.collection.convert.JavaCollectionWrappers$JPropertiesWrapper"),
     ProblemFilters.exclude[MissingTypesProblem]("scala.collection.convert.JavaCollectionWrappers$JSetWrapper"),
     ProblemFilters.exclude[MissingTypesProblem]("scala.collection.mutable.WeakHashMap"),
+
+    // Fix for scala/bug#12059
+    ProblemFilters.exclude[MissingClassProblem]("scala.collection.MapView$Keys"),
+    ProblemFilters.exclude[MissingClassProblem]("scala.collection.MapView$Values"),
```

Comment:
Wouldn't it be better to avoid making these jvm-public classes by creating anonymous inner classes?

Reply:
I don't know. I see your point, but at the same time, I think it's cleaner the way it is. I'd love to know what others think cc @julienrf

Reply:
I have a slight preference for having (private) named classes